src/icons/badges/SharedLinkBadge.js

```diff
@@ -3,42 +3,57 @@ import * as React from 'react';
 
 import { BOX_BLUE } from '../../common/variables';
 import AccessibleSVG from '../accessible-svg';
+import type { Icon, TwoTonedOutlinedStyle } from '../flowTypes';
 
-type Props = {
-    className?: string,
-    color?: string,
-    height?: number,
-    /** A text-only string describing the icon if it's not purely decorative for accessibility */
-    title?: string | React.Element<any>,
-    width?: number,
+type Props = Icon & {
+    styleProps?: TwoTonedOutlinedStyle,
 };
 
-const SharedLinkBadge = ({ className = '', color = BOX_BLUE, height = 16, title, width = 16 }: Props) => (
-    <AccessibleSVG
-        className={`shared-link-badge ${className}`}
-        height={height}
-        title={title}
-        viewBox="0 0 16 16"
-        width={width}
-    >
-        <g fill="none" fillRule="evenodd">
-            <path
-                className="fill-color"
-                d="M15 8c0-3.866-3.134-7-7-7S1 4.134 1 8s3.134 7 7 7 7-3.134 7-7zM0 8c0-4.418 3.582-8 8-8s8 3.582 8 8-3.582 8-8 8-8-3.582-8-8z"
-                fill={color}
+const SharedLinkBadge = ({
+    className = '',
+    color,
+    height = 16,
+    styleProps = {
```

Comment:
Mmm yeah, good point about spreading the values, I'll fix.

I originally had more specific names that mapped directly to the SVG (like `fill`, etc), but the problem is many icons have many different paths, circles, groups, etc with different fills and strokes that aren't always globally style-able. Without some sort of abstraction, this would open it up to infinite combinations. Imagine the chat icon, which has a path with a configurable fill color for the bubble and a path that's not configurable (at least not yet) for the dots inside the bubble. Would you then allow `bubbleFill`, `bubbleStroke`, `dotsFill`, and `dotsStroke`? Just declaring `fill` wouldn't be specific enough.

I think if we feel we need that much granularity that wouldn't be so bad, but some other folks gave feedback earlier that they'd like to see a common set of basic props that allow a basic set of shared concepts across all icon groups. So for example you'd have some icons that are "outline-able", i.e. you could set an outer border of a configurable width and color. And another class of icons might allow configurable outlines and configurable foreground/background colors (it would be up to the developer/designer to determine what is the foreground and what is the background for that particular type of icon). This was the approach we landed on due to the complexities mentioned above. Make sense @jstoffan?

Reply:
I understand. We do currently have a wide variety of icons and their styling can be quite different. My concern is that this approach could lead to an explosion of styling micro-formats (ThreeToneStyle, FilledStyle, InvertedStyle, etc.) and make it very difficult to ever standardize. 

Limiting ourselves to the properties I listed above would provide the same basic styling options without that risk. In the IconChat example, `stroke` could be used for the bubble outline and `fill` for its background color. We would only need something like `bubbleStroke` if we want to color the inner dots separately, which doesn't seem to be a requirement today. I would argue that if such a requirement is introduced, we should push back on it in favor of standardization.

Reply:
@jstoffan I think maybe we're trying to solve the same problem in different ways. The intention of creating a few base flow types was to **prevent** an explosion of styling formats or property names. The two base styles (`OutlinedStyle` and `TwoTonedStyle`) are sort of like mix-ins, and they have a fixed set of property names. Some icon families would use only the `OutlinedStyle` (like folder and file icons). And then TwoTonedOutlinedStyle was provided as a convenience since many of the icons need both of the mix-ins. In talking with design, this is likely to be the extent of the styling needs for the near future.

I believe you're suggesting a similar thing but perhaps without the notion of mix-ins, and with different property names (`stroke` and `fill` vs. `backgroundColor` and `foregroundColor`)? I had originally started with names like `stroke` and `fill` but moved away from that because I felt like those names were confusing since there are multiple strokes and fills within any given SVG, and so that actually made it confusing to understand what the given effect was going to be. To me, `backgroundColor` conveys the net visual effect (it will set the color of the background portion of the SVG), regardless of where the color gets applied (for example, depending on the SVG, it could actually end up being set on a stroke, or there could be two different fills and it only sets the color of one fill). The point of these particular names were to give enough abstraction so that we could use common names across all icons while avoiding confusing situations where, say, a prop called `fill` was actually being applied to a stroke because of the nature of a particular SVG.